### DISCUSSION TITLE
about custom deploy

### DISCUSSION THREAD

**Discussion by Alvazz** [PhilDL/remix-gospel-stack#25] [Q&A]
Hi, PhilDL.

I use this command

`pnpm run build --filter=remix-app...`

Successfully generated two directories, in remix-app,

One is located in remix-app/build
The other is located in remix-app/public/build.

I bought a vps server of vultr, and installed centos 8 system, and then installed npm, and pm2 tools.

I created a test-web directory under the root directory, and uploaded the index.js under the remix-app/build directory to test-web, and then used pm2 start index.js. Then I tried to access it through the ip address, but I couldn't access it successfully. Is my deployment process correct?

**Comment by Alvazz** [comment]
An extra thing to mention is that I'm not using 'Prisma', I'm just trying to purely display the page.

**Comment by PhilDL** [comment]
Hello,

This is gonna be tough to debug, in a monorepo with `pnpm` when you run the build command, your generated `server.js` file will still have references to node_modules with the path located in your monorepo. 
Trying to run `server.js` directly anywhere else than from the root of the monorepo will get you a path error I think.

I remember that it was hard to debug when I was creating the Dockerfile but that config is working right now, and I would suggest you fall back to Docker if you can't make it work with pm2.
But I think in general you should have the same steps as in the Dockerfile, (for the files/folders to COPY), I personally can't help you with pm2 as I didn't try that setup sorry.

**Comment by Jarrodsz** [comment]
I run the build command inside an apps folder. Each app folder has its own git repro with dokku paas as remote setup. I then git push dokku master and it automatically deploys my app with a dockfile for each app folder in its root. 

I also extended the packages to be placed in subfolders so /packages/core/auth packages/core/mailer and packages/marketing ( for landing page and blog code components etc)


I also have a seperate git setup in the root of the mono repro to just commit code turbo repro wide.

In my case i think this is the ideal setup not sure if there are any chaveats but this seems "best practice" to me. Could work for you too.

Any thoughts on my way of deploying this way?

**Comment by Alvazz** [comment]
> Hello,
> 
> This is gonna be tough to debug, in a monorepo with `pnpm` when you run the build command, your generated `server.js` file will still have references to node_modules with the path located in your monorepo. Trying to run `server.js` directly anywhere else than from the root of the monorepo will get you a path error I think.
> 
> I remember that it was hard to debug when I was creating the Dockerfile but that config is working right now, and I would suggest you fall back to Docker if you can't make it work with pm2. But I think in general you should have the same steps as in the Dockerfile, (for the files/folders to COPY), I personally can't help you with pm2 as I didn't try that setup sorry.

Thanks!

**Comment by Alvazz** [comment]
> I run the build command inside an apps folder. Each app folder has its own git repro with dokku paas as remote setup. I then git push dokku master and it automatically deploys my app with a dockfile for each app folder in its root.
> 
> I also extended the packages to be placed in subfolders so /packages/core/auth packages/core/mailer and packages/marketing ( for landing page and blog code components etc)
> 
> I also have a seperate git setup in the root of the mono repro to just commit code turbo repro wide.
> 
> In my case i think this is the ideal setup not sure if there are any chaveats but this seems "best practice" to me. Could work for you too.
> 
> Any thoughts on my way of deploying this way?

Can you share an example of your configuration?

**Comment by Jarrodsz** [comment]
> > I run the build command inside an apps folder. Each app folder has its own git repro with dokku paas as remote setup. I then git push dokku master and it automatically deploys my app with a dockfile for each app folder in its root.
> > I also extended the packages to be placed in subfolders so /packages/core/auth packages/core/mailer and packages/marketing ( for landing page and blog code components etc)
> > I also have a seperate git setup in the root of the mono repro to just commit code turbo repro wide.
> > In my case i think this is the ideal setup not sure if there are any chaveats but this seems "best practice" to me. Could work for you too.
> > Any thoughts on my way of deploying this way?
> 
> Can you share an example of your configuration?

Follow the dokku deploy instructions on there website its easy
